prql-compiler/src/lib.rs

```diff
@@ -105,28 +106,83 @@ pub static PRQL_VERSION: Lazy<Version> =
 /// Use the prql compiler to convert a PRQL string to SQLite dialect
 ///
 /// ```
-/// use prql_compiler::{compile, sql};
+/// use prql_compiler::{compile, Options, Target, sql::Dialect};
 ///
 /// let prql = "from employees | select [name,age] ";
-/// let opt = sql::Options {
+/// let opt = Options {
 ///     format: false,
-///     dialect: Some(sql::Dialect::SQLite),
+///     target: Target::Sql(Some(Dialect::SQLite)),
 ///     signature_comment: false
 /// };
-/// let sql = compile(&prql, Some(opt)).unwrap();
+/// let sql = compile(&prql, opt).unwrap();
 /// println!("PRQL: {}\nSQLite: {}", prql, &sql);
 /// assert_eq!("SELECT name, age FROM employees", sql)
 ///
 /// ```
 /// See [`sql::Options`](sql/struct.Options.html) and [`sql::Dialect`](sql/enum.Dialect.html) for options and supported SQL dialects.
-pub fn compile(prql: &str, options: Option<sql::Options>) -> Result<String, ErrorMessages> {
+pub fn compile(prql: &str, options: Options) -> Result<String, ErrorMessages> {
```

Comment:
@aljazerzen I don't have a strong view on whether we take an `Option` or force folks to provide a `Default`, but was this intentional?

Reply:
I realized that this was a bad api design, because None here has no semantic value since we have `Options::default()` anyways.